carnival/carnival.ts

```diff
@@ -0,0 +1,1017 @@
+            }
+
+        } else if (scoreType == scoreTypes.LScore) {
+            bestScore = settings.readNumber("low-score");
+            if (bestScore == undefined) { bestScore = Infinity; }
+            if (thisBest < bestScore) {
+                newBest = true;
+                bestScore = thisBest;
+                info.setScore(thisBest);
+                saveLowScore(thisBest);
+            }
+
+        } else if (scoreType == scoreTypes.HTime) {
+
+            // Set thisBest to timeElapsed if no customScore
+            if (!customScore) {
+                thisBest = timeElapsed;
+            }
+
+            if (thisBest > bestScore) {
+                newBest = true;
+                bestScore = thisBest;
+                info.setScore(thisBest);
+                info.saveHighScore();
+            }
+
+        } else if (scoreType == scoreTypes.LTime) {
+            bestScore = settings.readNumber("low-score");
+            if (bestScore == undefined) { bestScore = Infinity; }
+
+            // Set thisBest to timeElapsed if no customScore
+            if (!customScore) {
+                thisBest = timeElapsed;
+            }
+
+            if (thisBest < bestScore) {
+                newBest = true;
+                bestScore = thisBest;
+                info.setScore(thisBest);
+                saveLowScore(thisBest);
+            }
+
+        } else {
+
+            // Score judging type must be "None"
+            thisBest = undefined;
+            bestScore = undefined;
+            newBest = false;
+        }
+
+        // Make sure there's a sound to playe
+        if (!winSound) { winSound = music.powerUp; }
+
+        // releasing memory and clear fibers. Do not add anything that releases the fiber until background is set below,
+        // or screen will be cleared on the new frame and will not appear as background in the game over screen.
+        game.popScene();
+        game.pushScene();
+        scene.setBackgroundImage(screen.clone());
+
+        winSound.play();
+
+        fanfare.startScreenEffect();
+
+        pause(400);
+
+        const overDialog = new GameOverDialog(true, message, thisBest, bestScore, newBest);
+        scene.createRenderable(scene.HUD_Z, target => {
+            overDialog.update();
+            target.drawTransparentImage(
+                overDialog.image,
+                0,
+                (screen.height - overDialog.image.height) >> 1
+            );
+        });
+        pause(500); // wait for users to stop pressing keys
+        overDialog.displayCursor();
+        game.waitAnyButton();
+        control.reset();
+
+    }
+
+    function roundOff(thisNum: number, toPlace: number): number {
+        const x = Math.pow(10, toPlace);
+        return Math.round(thisNum * x) / x;
+    }
+
+    function init(winStyle: winTypes, fanfare: effects.BackgroundEffect) {
+        if (countdownInitialized) return;
+        countdownInitialized = true;
+
+        info.onCountdownEnd(function () {
```

Comment:
this is going to override the user's onCountdownEnd, right?

Reply:
I'm honestly not sure…but if they want to use their own onCountdownEnd, they should probably use the other countdown block.

Reply:
@riknoll Should I do something to make sure it does/doesn't?

Reply:
it might end up being confusing to the user if they try to use this block, but you'd basically need to re-implement the whole info category to avoid it. let's just leave it as is